apps/scan/src/main/scala/org/lfdecentralizedtrust/splice/scan/admin/http/HttpScanHandler.scala

```diff
@@ -833,6 +842,110 @@ class HttpScanHandler(
     }
   }
 
+  def getEventById(
+      updateId: String,
+      encoding: definitions.DamlValueEncoding,
+      extracted: TraceContext,
+  ): Future[Either[definitions.ErrorResponse, definitions.EventHistoryItem]] = {
+    implicit val tc = extracted
+    for {
+      eventO <- eventStore.getEventByUpdateId(updateId)
+    } yield {
+      eventO match {
+        case None => Left(definitions.ErrorResponse(s"Event with id $updateId not found"))
+        case Some((verdictWithViewsO, updateO)) =>
+          val encodedUpdateV2 = updateO
+            .map(ScanHttpEncodings.encodeUpdate(_, encoding, ScanHttpEncodings.V1))
+            .map(toUpdateV2)
+          val verdictEncoded = verdictWithViewsO.map { case (v, views) =>
+            ScanHttpEncodings.encodeVerdict(v, views)
+          }
+          Right(definitions.EventHistoryItem(encodedUpdateV2, verdictEncoded))
+      }
+    }
+  }
+
+  override def getEventById(respond: ScanResource.GetEventByIdResponse.type)(
+      updateId: String,
+      damlValueEncoding: Option[definitions.DamlValueEncoding],
+  )(extracted: TraceContext): Future[ScanResource.GetEventByIdResponse] = {
+    implicit val tc = extracted
+    withSpan(s"$workflowId.getEventById") { _ => _ =>
+      val encoding = damlValueEncoding.getOrElse(definitions.DamlValueEncoding.CompactJson)
+      getEventById(
+        updateId = updateId,
+        encoding = encoding,
+        extracted,
+      ).map {
+        case Left(error) => ScanResource.GetEventByIdResponse.NotFound(error)
+        case Right(update) => ScanResource.GetEventByIdResponse.OK(update)
+      }
+    }
+  }
+
+  def getEventHistory(
+      after: Option[definitions.UpdateHistoryRequestAfter] = None,
+      pageSize: Int,
+      encoding: definitions.DamlValueEncoding,
+      extracted: TraceContext,
+  ): Future[Vector[definitions.EventHistoryItem]] = {
+    implicit val tc: TraceContext = extracted
+    val updateHistory = store.updateHistory
+    val afterO = after.map { a =>
+      val afterRecordTime = parseTimestamp(a.afterRecordTime)
+      (a.afterMigrationId, afterRecordTime)
+    }
+
+    updateHistory
```

Comment:
Can you refactor this since now it repeats a lot of getUpdateHistory?

Reply:
Fixed